### DISCUSSION TITLE
SOC 2.9 so-rule-update not working

### DISCUSSION THREAD

**Discussion by ITgeevoSOC** [Security-Onion-Solutions/securityonion#6485] [Unsupported Versions] [answered]
Hi,

We have setup 2 security onions servers on our premises for testing. The first one is configured  to manage and monitoring through Public IP and second through private IP under FW. On the first SOC server the so-rule-update is working perfectly. On the second SOC server the so-rule-update is not working. The server has internet and also using the wget command is connected with out any issues. Also check the firewall and the rule is not blocked.
After checking the logs we found an error [ERRCODE: SC ERR NO RULES (42)] - No rule files match the pattern /etc/suricata/rules/all.rules.

We have check the suricata.yaml configuration on both servers and we both servers has as default path  /etc/suricata/rules but in the security onion this path doesn't exist. The rules are configured in the salt. 

Please note below attachments from both servers. 

Please can you advise how to reslove this issue?

Thanks in advance.
  
![SOC_Suricata logs file](https://user-images.githubusercontent.com/90396548/145211359-49c8b3ad-8846-445c-814d-eeb2ffe3ded9.PNG)
![SOC_Update rules working](https://user-images.githubusercontent.com/90396548/145211369-629e991a-8b49-42f3-8804-992392456983.PNG)
![SOC_Update rules not working](https://user-images.githubusercontent.com/90396548/145211375-28a69175-1d6d-4f32-a406-87c6b98bf829.PNG)
![suricata configuration on server that update rules are working](https://user-images.githubusercontent.com/90396548/145211378-bce7c1ef-8e5c-44cd-8104-e8dcc84c8e49.PNG)
![suricata configuration on server that update rules are not working](https://user-images.githubusercontent.com/90396548/145211383-c530631a-f360-4e6a-97eb-10a6a07eec62.PNG)
![SOC_wget command is working](https://user-images.githubusercontent.com/90396548/145211387-582ac479-0ba6-4f0b-a6c8-a7146967767b.PNG)

**Comment by dougburks** [comment]
For future reference, please provide text as actual text instead of screenshots.

When the machine tries to download the rules, is it going through any kind of HTTPS MITM?

**Reply by osotob** [reply]
Hi
I have same error.
Fresh Install version 2.3.91.
No events in soc. 
log from suricata.log:
22/12/2021 -- 11:21:59 - <Notice> - This is Suricata version 6.0.4 RELEASE running in SYSTEM mode
22/12/2021 -- 11:21:59 - <Info> - CPUs/cores online: 20
22/12/2021 -- 11:21:59 - <Info> - dropped the caps for main thread
22/12/2021 -- 11:21:59 - <Info> - eve-log output device (regular) initialized: /nsm/eve-%Y-%m-%d-%H:%M.json
22/12/2021 -- 11:21:59 - <Info> - stats output device (regular) initialized: stats.log
22/12/2021 -- 11:21:59 - <Info> - Running in live mode, activating unix socket
22/12/2021 -- 11:21:59 - <Warning> - [ERRCODE: SC_ERR_NO_RULES(42)] - No rule files match the pattern /etc/suricata/rules/all.rules
22/12/2021 -- 11:21:59 - <Warning> - [ERRCODE: SC_ERR_NO_RULES_LOADED(43)] - 1 rule files specified, but no rules were loaded!
22/12/2021 -- 11:21:59 - <Info> - Threshold config parsed: 0 rule(s) found
22/12/2021 -- 11:21:59 - <Info> - 0 signatures processed. 0 are IP-only rules, 0 are inspecting packet payload, 0 inspect application layer, 0 are decoder event only
22/12/2021 -- 11:21:59 - <Info> - Going to use 6 thread(s)
22/12/2021 -- 11:21:59 - <Info> - Running in live mode, activating unix socket
22/12/2021 -- 11:21:59 - <Info> - Using unix socket file '/var/run/suricata/suricata-command.socket'
22/12/2021 -- 11:21:59 - <Notice> - all 6 packet processing threads, 4 management threads initialized, engine started.
22/12/2021 -- 11:21:59 - <Info> - All AFP capture threads are running.
22/12/2021 -- 11:24:17 - <Notice> - Flow emergency mode entered...
22/12/2021 -- 11:24:24 - <Notice> - Flow emergency mode over, back to normal... unsetting FLOW_EMERGENCY bit (ts.tv_sec: 1640172264, ts.tv_usec:466188) flow_spare_q status(): 284% flows at the queue


When trying so-rule-update:


[sudo] password for adminsonda:
2021-12-22 11:33:24,016 - <INFO> - Loading ./rulecat.conf.
2021-12-22 11:33:24,020 - <INFO> - Forcing Suricata version to 6.0.
2021-12-22 11:33:24,020 - <INFO> - Fetching https://rules.emergingthreats.net/open/suricata-6.0.0/emerging.rules.tar.gz.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/urllib/request.py", line 1346, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/usr/local/lib/python3.9/http/client.py", line 1279, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.9/http/client.py", line 1325, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.9/http/client.py", line 1274, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.9/http/client.py", line 1034, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.9/http/client.py", line 974, in send
    self.connect()
  File "/usr/local/lib/python3.9/http/client.py", line 1448, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/usr/local/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/local/lib/python3.9/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.9/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/idstools-rulecat", line 12, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 845, in main
    files = Fetch(args).run()
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 329, in run
    files.update(self.fetch(url))
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 317, in fetch
    idstools.net.get(
  File "/usr/local/lib/python3.9/site-packages/idstools/net.py", line 55, in get
    remote = urlopen(url)
  File "/usr/local/lib/python3.9/urllib/request.py", line 214, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python3.9/urllib/request.py", line 517, in open
    response = self._open(req, data)
  File "/usr/local/lib/python3.9/urllib/request.py", line 534, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/local/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python3.9/urllib/request.py", line 1389, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/usr/local/lib/python3.9/urllib/request.py", line 1349, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)>


My box is behind a proxy. I installed cert to be able to install and surf.
Any help please?

**Comment by ITgeevoSOC** [comment, marked as answer]
Hi, 

Sorry for the delay reply.

I found that the issue is maybe on the MTU. The MTU in my private network is 1450, but the MTU in dockers are 1500.
please note the link below. 

https://serverfault.com/questions/1018266/hetzner-private-network-1-docker-host-1-pfsense-i-keep-getting-tcp-previous

Can we use the same proceture to change the Security Onion MTU on the docker-containers?

Thanks in advance

**Reply by dougburks** [reply]
Looks like you've created a separate discussion for this:
https://github.com/Security-Onion-Solutions/securityonion/discussions/6687#discussioncomment-1863964
